### PR TITLE
Network refactor

### PR DESCRIPTION
**Pull request by mfornet**
Use routed messages instead of peer messages whenever necessary.

- [x] StateRequest
- [x] ChunkPart
- [x] ChunkOnePart
- [x] Fix `TxStatusRequest` handling of routing back messages.
- ~Block~
- ~BlockHeader~

### CODE REVIEW

File: chain/chunks/src/lib.rs [hunk 1]
```diff
@@ -142,8 +142,8 @@ pub struct ShardsManager {
     requested_one_parts: RequestPool,
     requested_chunks: RequestPool,
 
-    requests_fifo: VecDeque<(ShardId, ChunkHash, PeerId, u64)>,
-    requests: HashMap<(ShardId, ChunkHash, u64), HashSet<(PeerId)>>,
+    requests_fifo: VecDeque<(ShardId, ChunkHash, CryptoHash, u64)>,
+    requests: HashMap<(ShardId, ChunkHash, u64), HashSet<(CryptoHash)>>,
```
Comment (bowenwang1996) [inline chain/chunks/src/lib.rs]: Why do you have parenthesis around `CryptoHash`?

File: chain/network/src/peer.rs [hunk 2]
```diff
@@ -263,47 +265,54 @@ impl Peer {
             PeerMessage::BlockHeaders(headers) => {
```
Comment (bowenwang1996) [inline chain/network/src/peer.rs]: Why is block header not routed?

Comment (mfornet) [reply chain/network/src/peer.rs]: Discussed with @SkidanovAlex . It is not needed for Blocks/BlockHeaders, you should be able to get them from neighboring peers, no need to request them from nodes farther.

File: chain/chunks/src/lib.rs (hunk 1 above)
Comment (mfornet) [reply chain/chunks/src/lib.rs]: It was like this before I get my hands into this, but yes, I think this was for no good reason as I see in the code, will change it.

**Pull request merged**